interfaces/cython/cantera/examples/onedim/convolute.pyx

```diff
@@ -0,0 +1,160 @@
+        void Close()   
+         
+    cdef cppclass CreateDiffusionFlameletsFile:
+        CreateDiffusionFlameletsFile() except +
+        CreateDiffusionFlameletsFile(char*) except +
+        void WriteHeader(double, int, int, double) except +
+        void WriteData(int, double*, char*) except +
+        void WriteDataMin(int, double*, char*, double) except +
+        void WriteLaminarDataReaction(int, int, double*, int, double*, int, int*) except +
+        void WriteTurbulentDataReaction(int, int, double*, int, double*, int, int*, double) except +
+        void WriteDataProduction(int, double*, double, char*) except +
+        void WriteCR() except +
+        void Close() except +
+
+cdef class PyConvolute:
+    cdef ConvoluteCpp c_conv      # hold a C++ instance which we're wrapping
+    def __cinit__(self):
+        self.c_conv = ConvoluteCpp()
+    def __cinit__(self, name):
+        cdef py_bytes = name.encode()
+        cdef char* c_name = py_bytes
+        self.c_conv = ConvoluteCpp(c_name)
+    def set_betas(self, np.ndarray[double, ndim=3, mode="c"] input not None):
+        len = input.shape
+        self.c_conv.SetBetas(len[0], len[1], len[2], &input[0,0,0])
+        return True
+    def set_betas2(self, np.ndarray[double, ndim=3, mode="c"] input not None):
+        len = input.shape
+        self.c_conv.SetBetas2(len[0], len[1], len[2],  &input[0,0,0])
+        return True
+    def set_dict_entry(self, np.ndarray[double, ndim=2, mode="c"] input not None):
+        len = input.shape
+        self.c_conv.SetDictEntry(len[0], len[1], &input[0,0])
+        return True
+    def set_zz(self, np.ndarray[double, ndim=2, mode="c"] input not None):
+        len = input.shape
+        self.c_conv.SetZz(len[0], len[1], &input[0,0])
+        return True
+    def set_zx(self, np.ndarray[double, ndim=1, mode="c"] input not None):
+        len = input.shape
+        self.c_conv.SetZx(len[0], &input[0])
+        return True
+    def set_zy(self, np.ndarray[double, ndim=1, mode="c"] input not None):
+        len = input.shape
+        self.c_conv.SetZy(len[0], &input[0])
+        return True
+    def eval_normal(self, cvar, zvar, zmeanIndex, zvarIndex, cmeanIndex, cvarIndex):
+        self.c_conv.EvalNormal(cvar, zvar, zmeanIndex, zvarIndex, cmeanIndex, cvarIndex)
+    def eval_rho(self, cvar, zvar, zmeanIndex, zvarIndex, cmeanIndex, cvarIndex):
+        self.c_conv.EvalRho(cvar, zvar, zmeanIndex, zvarIndex, cmeanIndex, cvarIndex)
+    def eval(self, cvar, zvar, zmeanIndex, zvarIndex, cmeanIndex, cvarIndex):
+        self.c_conv.Eval(cvar, zvar, zmeanIndex, zvarIndex, cmeanIndex, cvarIndex)
+    def eval_zz_array(self):
+        self.c_conv.EvalZzArray()
+    def eval_zx_array(self):
+        self.c_conv.EvalZxArray()
+    def eval_beta_array(self):
+        self.c_conv.EvalBetaArray()
+    
+cdef class PyCreateOutputFile:
+    cdef CreateOutputFile c_conv      # hold a C++ instance which we're wrapping

```

Comment:
There should be 2 spaces between the code and the inline comment

    cdef CreateOutputFile c_conv  # hold a C++ instance which we're wrapping
                                ^^

but is this comment needed?